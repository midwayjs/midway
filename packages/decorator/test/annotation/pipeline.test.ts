import {
  Pipeline,
  Provide,
  getPropertyInject,
  getConstructorInject,
  PIPELINE_IDENTIFIER,
  NAMED_TAG,
} from '../../src';

@Provide()
class Test {
  constructor(@Pipeline() tt: any) {
    // ignore
  }

  @Pipeline()
  dd: any;
}

describe('/test/annotation/pipeline.test.ts', () => {
  it('pipeline decorator should be ok', () => {
    const p = getPropertyInject(Test);
    expect(p).toEqual({
      dd: [
        {
          args: undefined,
          key: 'inject',
          value: PIPELINE_IDENTIFIER,
        },
      ],
    });

    expect(p['dd'][0].toString()).toEqual(
      'tagged: { key:inject, value: __pipeline_identifier__ }'
    );

    const c = getConstructorInject(Test);
    expect(c).toEqual({
      0: [
        {
          args: undefined,
          key: 'inject',
          value: PIPELINE_IDENTIFIER,
        },
      ],
    });

    expect(c[0].toString()).toEqual(
      'tagged: { key:inject, value: __pipeline_identifier__ }'
    );

    const meta = c[0][0];
    meta.key = NAMED_TAG;
    expect(meta.toString()).toEqual('named: __pipeline_identifier__ ');
  });
});
