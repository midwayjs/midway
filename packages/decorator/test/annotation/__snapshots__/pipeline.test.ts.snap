// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/annotation/pipeline.test.ts pipeline decorator should be ok 1`] = `
{
  "dd": {
    "impl": true,
    "key": "__pipeline_identifier__",
    "metadata": {
      "valves": undefined,
    },
    "propertyName": "dd",
  },
}
`;
